superduperdb/base/superduper.py

```diff
@@ -24,24 +24,13 @@ def superduper(item: t.Optional[t.Any] = None, **kwargs) -> t.Any:
 
 
 def _auto_identify_connection_string(item: str, **kwargs) -> t.Any:
-    from superduperdb.base.build import build_datalayer
-
-    if item.startswith('mongomock://'):
-        kwargs['data_backend'] = item
+    if re_match(r'^[a-zA-Z0-9]+://', item) is None:
+        raise ValueError(f'{item} is not a valid connection string')
 
-    elif item.startswith('mongodb://'):
-        kwargs['data_backend'] = item
+    kwargs['data_backend'] = item
```

Comment:
Hi @ChandanChainani thanks for the PR. Are you sure that these can be dropped like this? Do you get passing tests?

Reply:
Hi @blythed, Yes, except for llm and doc_string related test all the other tests are validated
### Command
```
pytest ./test/unittest/ -svv --ignore=test/unittest/ext/test_llama_cpp.py --ignore=test/unittest/ext/llm --ignore=test/unittest/ext/transformers/test_llm.py --ignore=test/unittest/ext/transformers/test_llm_training.py --ignore=test/unittest/ext/transformers/test_transformers.py
```
### Output
```
=============================================== short test summary info ===============================================
FAILED test/unittest/cli/test_cli.py::test_cli_info - subprocess.CalledProcessError: Command '('python', '-m', 'superduperdb', 'info')' returned non-zero exit status 1.
ERROR test/unittest/backends/ibis/test_query.py::test_renamings - NotADirectoryError: [WinError 2607] The directory name is invalid: 'C:\\Users\\...\\AppData\\Local\\Temp\\tmphf3n65j3\\test.ddb'
=========================== 1 failed, 187 passed, 1110 skipped, 1 error in 82.56s (0:01:22) ===========================
```

Reply:
@blythed

I have update the code and there is one condition that I think can be added in the code to make sure the user only supply supported db/file uri format please suggest (for supported db/file format we can maintain constants)